LocationUpdatesBackgroundKotlin/app/src/main/java/com/google/android/gms/location/sample/locationupdatesbackgroundkotlin/LocationUpdatesBroadcastReceiver.kt

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (C) 2020 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.gms.location.sample.locationupdatesbackgroundkotlin
+
+import android.app.ActivityManager
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.util.Log
+import com.google.android.gms.location.LocationResult
+import com.google.android.gms.location.sample.locationupdatesbackgroundkotlin.data.LocationRepository
+import com.google.android.gms.location.sample.locationupdatesbackgroundkotlin.data.db.MyLocationEntity
+import java.util.Date
+import java.util.concurrent.Executors
+
+private const val TAG = "LUBroadcastReceiver"
+
+/**
+ * Receiver for handling location updates.
+ *
+ * For apps targeting API level O and above
+ * {@link android.app.PendingIntent#getBroadcast(Context, int, Intent, int)} should be used when
+ * requesting location updates in the background. Due to limits on background services,
+ * {@link android.app.PendingIntent#getService(Context, int, Intent, int)} should NOT be used.
+ *
+ *  Note: Apps running on "O" devices (regardless of targetSdkVersion) may receive updates
+ *  less frequently than the interval specified in the
+ *  {@link com.google.android.gms.location.LocationRequest} when the app is no longer in the
+ *  foreground.
+ */
+class LocationUpdatesBroadcastReceiver : BroadcastReceiver() {
+
+    override fun onReceive(context: Context, intent: Intent) {
+        Log.d(TAG, "onReceive() context:$context, intent:$intent")
+
+        if (intent.action == ACTION_PROCESS_UPDATES) {
+            val result = LocationResult.extractResult(intent)
+            if (result != null) {
+                for (location in result.locations) {
```

Comment:
`result.locations.foreach...`

also consider using `LocationResult.extractResult(intent).apply{ ... }`

Reply:
Switched!

I used `let` instead for the null check, which _I think_ is what you meant? I only ever use `apply` for object configuration... but you probably have a jedi way for using it, so I will followup and fix in next PR if that is the case. :)